src/content/docs/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,81 @@
+
+## Semantic versioning
+
+Astro attempts to adhere as much as possible to [semantic versioning](https://semver.org/), which is a set of rules developers use to determine how to assign a version number to a release. Semantic version follows predictable to inform users of the kind of changes they can expect from one version to the next.
+
+Semantic versioning enforces a pattern of `X.Y.Z` for software version numbers. These values represent **major (X)**, **minor (Y)**, and **patch (Z)** updates.
+
+### Patch changes
+
+Patch changes are the least disruptive changes. They do not change the way you use Astro, and no change to your own code is required when you update.
+
+When Astro issues a "patch" version, the last number increases. (e.g. `astro@4.3.14` -> `astro@4.3.15`)
+
+Patches may be released for reasons such as:
+
+- Re-releases after a failed release.
+- Internal changes that do not change Astro’s functionality:
+  - refactors
+  - performance improvements
+  - increase or change in test coverage
+  - aligning with stated documentation and expected behavior
+- Improvements to logging and error messages.
+
+Patch changes also include bug fixes, even in cases where users were taking advantage of existing unintended or undesirable behavior.
+
+
+### Minor changes
+
+
+Minor releases primarily introduce new features and improvements that you may wish to try, but require no changes to your code. Some existing features may also be **deprecated** (marked for deletion in a future version while continuing to function) in a minor release, giving you the opportunity to prepare for their eventual removal.
+
+Minor releases include changes such as:
+
+- **Deprecations** of existing features/options.
+- Introduction of new functionalities.
+- Upgrade of minimal required Node.js version (minor or patch), e.g. from `v18.4.0` to `v18.7.0`.
+  :::note
+  We take these kind of changes very seriously, and approve them only if we think that a new version of Node.js can improve the DX of our users, and impact core functionalities.
+  :::
+- Introduction of new options in the integration hooks.
+- Introduction of new functionalities in `astro/app`, notably used for creating new adapters.
+
+A minor release may also include smaller, patch changes at the same time.
+
+### Major changes
+
+Major releases will include breaking changes to at least some existing code. These breaking changes are always documented in an "Upgrade to vX" guide in Astro.
+
+Major releases allow Astro to make significant changes not only to internal logic, but also in intended behavior and usage. Documentation will be updated to reflect the latest version only, and a **static, unmaintained snapshot of older docs** are available as a historical record for older projects that are not yet upgraded.
+
+Major releases include changes such as:
+
+- Upgrade of minimal required Node.js version (minor or major), e.g. from `v18.4.0` to `v20.0.0`.
```

Comment:
A minor bump in Node can be both a minor and a major for us?

Reply:
Yes. That's the controversial part that started the discussion, which ended up with the creation of this document. Usually, we want to try to keep Node.js **minor** bumps to a major release, but some situations came up that we felt needed clarification.

For example, the latest version of `sharp` requires a Node.js version that is more recent than our minimal version. Updating to that `sharp` version would allow us to fix some nasty workarounds and bugs.

Reply:
I don't see any problem in bumping a minor in a minor in general. A minor version guarantees no breaking changes in the contract between the consumer and the thing being versioned. And that is upheld when bumping a dependency, one upgrade requiring another and when both are applied and both are minor the consumer project should not break.

We can't make guarantees about anything else in the project, just between the project and Astro itself. A bump in a patch version of a dependency can be a breaking change because some other dependency outside of Astro may no longer be de-duplicated and it's types won't match up. That doesn't affect the semver guarantees of Astro, it is the same as relying on an implementation detail